srcpkgs/starship/template

```diff
@@ -11,8 +11,25 @@ maintainer="Aluísio Augusto Silva Gonçalves <aluisio@aasg.name>"
 license="ISC"
 homepage="https://starship.rs"
 distfiles="https://github.com/starship/starship/archive/v${version}.tar.gz"
-checksum=ce2c86e3f3a7d8cd2e1a204e7ef491473fdd93c41bb43a4718fdd66039f7bc99
+checksum=cf7b5848949bc1e61dc771ef2a429e4ccb8d339e1ca705bdf521fc4a4fc96309
+
+if [ "$CROSS_BUILD" ]; then
+	hostmakedepends+=" starship"
+fi
 
 post_install() {
+	STARSHIP="target/${RUST_TARGET}/release/starship"
+	if [ "$CROSS_BUILD" ]; then
+		STARSHIP=starship
+	fi
+	# generate shell completions
+	${STARSHIP} completions zsh > starship.zsh
+	${STARSHIP} completions bash > starship.bash
+	${STARSHIP} completions fish > starship.fish
```

Comment:
I'm in favour of moving those completion generation to build stage.
post_build, perhaps.

Reply:
What do you think of this https://github.com/ericonr/void-packages/blob/vcompletions/common/environment/setup/install.sh#L262 ?

Reply:
I could split it up too, of course

Reply:
> What do you think of this https://github.com/ericonr/void-packages/blob/vcompletions/common/environment/setup/install.sh#L262 ?

I don't know. A bit too much magic :)

Anyway, most packages ships the completion instead of depending on some generator.

Or is it the new trend, every packages started to ship completion generated from command?

Thinking about this a bit, I wonder why do we need to ship it.
We can generate the completions in post-update hook (`INSTALL` file),
this will avoid the whole pulling host version,
(some generated completion maybe wrong, let's say something similar to `-march` of `gcc`).
Yeah, I probably add a new xbps-trigger type, let's say completion generator, feed it a cmd name.

For you suggested function, (even if I don't like it), If I were you, I might go with:

```sh
test -f $cmd.bash && _vinstall ${cmd}.bash 0644 $_bash_completion_dir $cmd
test -f $cmd.fish && _vinstall ${cmd}.fish 0644 $_fish_completion_dir ${cmd}.fish
test -f $cmd.zsh && _vinstall ${cmd}.zsh 0644 $_zsh_completion_dir _${cmd}
```

Reply:
I'm rushing for `$dayjobs` now, maybe I was plain wrong about above comment :)

Reply:
> Or is it the new trend, every packages started to ship completion generated from command? 

Might be, since they are using frameworks for the command line stuff, which also generate the completions. I wish they could ship tarballs with pre generated ones, but that would be a bit hard in the current GiitHub release mechanism where you just tag a specific version.

I like the idea of a custom hook, but it has some issues:

- some packages use `completions`, others `completion`, for the subcommand. 
- `rustup-init` (the binary shipped in `rustup`), for example, needs to be symlinked to `rustup`, otherwise it errors out when generating completions (yes, I hate it too).
- some packages might not support all shells.

I really like your suggestion for my function, though! It would make stuff less magical, but not by much :p . Thanks :)

Reply:
I think we can merge it as is after fixing test failures for now.
We'll take #22785 as discussion for the completion generator :)